controllers/install/lookout_controller.go

```diff
@@ -0,0 +1,219 @@
+		}
+	}
+
+	if components.Service != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.Service, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	// now do init logic
+
+	return ctrl.Result{}, nil
+}
+
+type LookoutComponents struct {
+	ClusterRole        *rbacv1.ClusterRole
+	ClusterRoleBinding *rbacv1.ClusterRoleBinding
+	Ingress            *networking.Ingress
+	IngressRest        *networking.Ingress
+	Deployment         *appsv1.Deployment
+	Service            *corev1.Service
+	ServiceAccount     *corev1.ServiceAccount
+	Secret             *corev1.Secret
+}
+
+func generateLookoutInstallComponents(lookout *installv1alpha1.Lookout, scheme *runtime.Scheme) (*LookoutComponents, error) {
+	secret, err := builders.CreateSecret(lookout.Spec.ApplicationConfig, lookout.Name, lookout.Namespace)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetOwnerReference(lookout, secret, scheme); err != nil {
+		return nil, err
+	}
+	deployment := createLookoutDeployment(lookout)
+	if err := controllerutil.SetOwnerReference(lookout, deployment, scheme); err != nil {
+		return nil, err
+	}
+	service := createLookoutService(lookout)
+	if err := controllerutil.SetOwnerReference(lookout, service, scheme); err != nil {
+		return nil, err
+	}
+	clusterRole := createLookoutClusterRole(lookout)
+	if err := controllerutil.SetOwnerReference(lookout, clusterRole, scheme); err != nil {
+		return nil, err
+	}
+
+	return &LookoutComponents{
+		Deployment:     deployment,
+		Service:        service,
+		ServiceAccount: nil,
+		Secret:         secret,
+		ClusterRole:    clusterRole,
+	}, nil
+}
+
+// Function to build the deployment object for Lookout.
+// This should be changing from CRD to CRD.  Not sure if generailize this helps much
+func createLookoutDeployment(lookout *installv1alpha1.Lookout) *appsv1.Deployment {
+	var replicas int32 = 1
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{Name: lookout.Name, Namespace: lookout.Namespace},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels:      nil,
+				MatchExpressions: nil,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{},
+				Spec:       corev1.PodSpec{},
+			},
+			Strategy:                appsv1.DeploymentStrategy{},
+			MinReadySeconds:         0,
+			RevisionHistoryLimit:    nil,
+			Paused:                  false,
+			ProgressDeadlineSeconds: nil,
+		},
+	}
+	return &deployment
+}
+
+func createLookoutService(lookout *installv1alpha1.Lookout) *corev1.Service {
+	service := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: lookout.Name, Namespace: lookout.Namespace},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Name:     "metrics",
+				Protocol: corev1.ProtocolTCP,
+				Port:     9001,
+			}},
+		},
+	}
+	return &service
+}
+
+func createLookoutClusterRole(lookout *installv1alpha1.Lookout) *rbacv1.ClusterRole {
```

Comment:
I don't think this is needed for lookout.  https://github.com/G-Research/armada/tree/master/deployment/lookout/templates

I don't see a ClusterRole.

Reply:
Yeah, I think I accidentally left this in from a executor merge earlier.